client/state/sites/plans/actions.js

```diff
@@ -122,7 +121,7 @@ export function fetchSitePlans( siteId ) {
  * @returns {Object} the corresponding action object
  */
 export function fetchSitePlansCompleted( siteId, data ) {
-	data = reject( data, '_headers' );
+	delete data._headers;
```

Comment:
I'll address this specific concern in a different pull request since it touches several files that are not necessary related to the checkout.